packages/swapper/src/swappers/ZrxSwapper/endpoints.ts

```diff
@@ -42,6 +42,8 @@ export const zrxApi: SwapperApi = {
     const { affiliateBps, receiveAddress, slippageTolerancePercentageDecimal, steps } = tradeQuote
     const { buyAsset, sellAsset, sellAmountIncludingProtocolFeesCryptoBaseUnit } = steps[0]
 
+    if (!receiveAddress) throw new Error('receiveAddress is required for ZrxSwapper quotes')
```

Comment:
just noticed, we are kind of missusing receive address for zrx. cross account is not supported and really should be `sendAddress` for all intents and purposes as used for `takerAddress` in the price/quote request. also just generally speaking it feels like it would make more sense to use `hasWallet` as the primary discriminator to make it more clear the intention imo.

Reply:
just noticed this is from `TradeQuote` vs `GetTradeQuoteInput` which does not include `hasWallet`. in this case I suppose it makes sense enough

Reply:
can use `isExecutableTradeQuote` though.

Reply:
> just noticed this is from `TradeQuote` vs `GetTradeQuoteInput` which does not include `hasWallet`. in this case I suppose it makes sense enough

Precisely, while this looks like a big diff at first glance, the intent is for a minimal changeset here and not try and change the underlying implementations of each swapper unless required to make it work without a wallet.

Agreed ZRX (and well, not only, CoW would be too) is weird terminology-wise, which is something I've noticed while working in this domain, feels weird that `receiveAddress` is the discriminator... but it's really what we use both for send side (if cross-account not supported, else receive address must be different) *and* receive side across the board, which is probably something we want to tackle in a follow-up 🙏🏽 

And nice catch on `isExecutableTradeQuote`! 486607c07f